src/AutoRest.CSharp/LowLevel/Output/OperationConfidenceChecker.cs

```diff
@@ -0,0 +1,196 @@
+                {
+                    foreach (var dm in type.DerivedModels)
+                    {
+                        visitedModels[dm] = confidenceLevel;
+                    }
+                }
+            }
+
+            visitedModels[type] = confidenceLevel;
+            return confidenceLevel;
+        }
+
+        private static ConvenienceMethodConfidenceLevel WalkListType(InputListType listType, TypeFactory typeFactory, Dictionary<InputModelType, ConvenienceMethodConfidenceLevel?> visitedModels)
+            => WalkType(listType.ElementType, typeFactory, visitedModels);
+
+        private static ConvenienceMethodConfidenceLevel WalkDictionaryType(InputDictionaryType dictType, TypeFactory typeFactory, Dictionary<InputModelType, ConvenienceMethodConfidenceLevel?> visitedModels)
+        {
+            var confidenceLevel = ConvenienceMethodConfidenceLevel.Confident;
+
+            var keyLevel = WalkType(dictType.KeyType, typeFactory, visitedModels);
+            UpdateConfidenceLevel(ref confidenceLevel, keyLevel);
+
+            var valueLevel = WalkType(dictType.ValueType, typeFactory, visitedModels);
+            UpdateConfidenceLevel(ref confidenceLevel, valueLevel);
+
+            return confidenceLevel;
+        }
+
+        private static ConvenienceMethodConfidenceLevel WalkLiteralType(InputLiteralType literalType, TypeFactory typeFactory)
+        {
+            // a literal type is not confident, when we wrap it wiht a number-valued enum without proper names for its enum value items
+            if (literalType.LiteralValueType is not InputEnumType enumType)
+                return ConvenienceMethodConfidenceLevel.Confident;
+
+            var isConfident = true;
+            var csharpType = typeFactory.CreateType(enumType);
+            if (csharpType is { IsFrameworkType: false, Implementation: EnumType @enum })
+            {
+                foreach (var value in @enum.Values)
+                {
+                    // this checks if the enum value is properly named. Numbers like `_1` or `_314` will return false, which is considered as "not properly named" thus not confident.
+                    if (IsUnreasonableName(value.Declaration.Name))
+                    {
+                        isConfident = false;
+                        break;
+                    }
+                }
+            }
+
+            return isConfident ? ConvenienceMethodConfidenceLevel.Confident : ConvenienceMethodConfidenceLevel.Internal;
+        }
+
+        private static ConvenienceMethodConfidenceLevel WalkUnionType(InputUnionType unionType)
+        {
+            // the union types are always not confident.
+            // if any exceptions happen in the future, we could handle it here.
+            return ConvenienceMethodConfidenceLevel.Internal;
+        }
+
+        private static bool IsUnreasonableName(string name) => name.StartsWith('_');
```

Comment:
The anonymous model name will be updated to `id`. so this function will not filter out anonymous model and set it as low confident. So where do you handle anonymous model?

And the model name which is not valid, such as keyword, the name will perfix `_`, so all those model are marked as low confident?

Reply:
> So where do you handle anonymous model?
I did it in the `TypeSpecInputModelTypeConverter`

Why I did not just use `inputModel.Name` and check if it is valid identifier?
I would like to make sure this could work with customization code, therefore I choose to check the `modelTypeProvider.Declaration.Name` instead of checking `InputModel.Name`.

Reply:
This method is now removed.
Instead, for model case, since anonymous model related things will be removed completely - there is no point to consider customizations (well in the meantime, they could always disable the post processor, and see those models, and introduce customization code, but since it relies on internally generated indices, it is not recommended)
for literal case, the check is:
first check if the name in input enum value is not valid identifier, then we compare the declaration name is the same as input enum value to ensure it has not been customized. If both match, we mark it as low confident.